{"ast":null,"code":"import fetch from 'isomorphic-fetch';\n\nconst postContactForm = () => {\n  return new Promise((resolve, reject));\n};\n\nexport function postContact(body, actions) {\n  console.log('body: ', JSON.stringify(body), body);\n  return fetch(\"http://localhost:3000/test/api/auth/contact\", {\n    method: 'POST',\n    headers: {\n      'Accept': 'application/json',\n      \"Content-Type\": \"application/json\"\n    },\n    cache: 'default',\n    mode: 'cors',\n    body: JSON.stringify(body)\n  }).then(response => {\n    return response;\n  }).catch(error => {\n    /* If your server returns something like a 422 when the\r\n       validation fails it will move to the catch part. Here\r\n       you can use the actions.setFieldError to set the errors\r\n       in your form.\r\n    */\n    error.response.data.forEach(error => {\n      actions.setFieldError(error.field, error.message);\n    });\n    return error;\n  }).finally(() => actions.setSubmitting(false)); //     .then(function (response) {\n  //     // var apiResponse;\n  //     // if (response.status == 422) {\n  //     //     apiResponse = response.errors;\n  //     // }\n  //     console.log('test 123', response.json());\n  //     // console.log('test 1 hier is he tnu', response.status);\n  //     return response;\n  // })\n  // .then(function (json) {\n  //     // console.log('test 2', JSON.stringify(json));\n  //     return json;\n  // }).catch(function(err){\n  //     console.log(err);\n  //     return err;\n  // });\n}","map":{"version":3,"sources":["C:/Projects/bedrijfswebsite-nextjs/api/contact/index.js"],"names":["fetch","postContactForm","Promise","resolve","reject","postContact","body","actions","console","log","JSON","stringify","method","headers","cache","mode","then","response","catch","error","data","forEach","setFieldError","field","message","finally","setSubmitting"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,kBAAlB;;AAEA,MAAMC,eAAe,GAAG,MAAM;AAC1B,SAAO,IAAIC,OAAJ,EAAaC,OAAO,EAAEC,MAAtB,EAAP;AACH,CAFD;;AAIA,OAAO,SAASC,WAAT,CAAsBC,IAAtB,EAA4BC,OAA5B,EAAqC;AACxCC,EAAAA,OAAO,CAACC,GAAR,CAAY,QAAZ,EAAsBC,IAAI,CAACC,SAAL,CAAeL,IAAf,CAAtB,EAA4CA,IAA5C;AAEA,SAAON,KAAK,CAAC,6CAAD,EAAgD;AACxDY,IAAAA,MAAM,EAAE,MADgD;AAExDC,IAAAA,OAAO,EAAE;AACL,gBAAS,kBADJ;AAEL,sBAAgB;AAFX,KAF+C;AAMxDC,IAAAA,KAAK,EAAE,SANiD;AAOxDC,IAAAA,IAAI,EAAE,MAPkD;AAQxDT,IAAAA,IAAI,EAAEI,IAAI,CAACC,SAAL,CAAeL,IAAf;AARkD,GAAhD,CAAL,CASJU,IATI,CASCC,QAAQ,IAAI;AAChB,WAAOA,QAAP;AACH,GAXM,EAYNC,KAZM,CAYAC,KAAK,IAAI;AACZ;;;;;AAKAA,IAAAA,KAAK,CAACF,QAAN,CAAeG,IAAf,CAAoBC,OAApB,CAA4BF,KAAK,IAAI;AACjCZ,MAAAA,OAAO,CAACe,aAAR,CAAsBH,KAAK,CAACI,KAA5B,EAAmCJ,KAAK,CAACK,OAAzC;AACH,KAFD;AAGA,WAAOL,KAAP;AACH,GAtBM,EAuBNM,OAvBM,CAuBE,MAAMlB,OAAO,CAACmB,aAAR,CAAsB,KAAtB,CAvBR,CAAP,CAHwC,CA4BxC;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACH","sourcesContent":["import fetch from 'isomorphic-fetch'\r\n\r\nconst postContactForm = () => {\r\n    return new Promise((resolve, reject))\r\n}\r\n\r\nexport function postContact (body, actions) {\r\n    console.log('body: ', JSON.stringify(body), body);\r\n\r\n    return fetch(\"http://localhost:3000/test/api/auth/contact\", {\r\n        method: 'POST',\r\n        headers: {\r\n            'Accept':'application/json',\r\n            \"Content-Type\": \"application/json\",\r\n        },\r\n        cache: 'default',\r\n        mode: 'cors',\r\n        body: JSON.stringify(body)\r\n    }).then(response => {\r\n        return response;\r\n    })\r\n    .catch(error => {\r\n        /* If your server returns something like a 422 when the\r\n           validation fails it will move to the catch part. Here\r\n           you can use the actions.setFieldError to set the errors\r\n           in your form.\r\n        */\r\n        error.response.data.forEach(error => {\r\n            actions.setFieldError(error.field, error.message)\r\n        })\r\n        return error;\r\n    })\r\n    .finally(() => actions.setSubmitting(false))\r\n\r\n    //     .then(function (response) {\r\n    //     // var apiResponse;\r\n    //     // if (response.status == 422) {\r\n    //     //     apiResponse = response.errors;\r\n    //     // }\r\n    //     console.log('test 123', response.json());\r\n    //     // console.log('test 1 hier is he tnu', response.status);\r\n    //     return response;\r\n    // })\r\n    // .then(function (json) {\r\n    //     // console.log('test 2', JSON.stringify(json));\r\n    //     return json;\r\n    // }).catch(function(err){\r\n    //     console.log(err);\r\n    //     return err;\r\n    // });\r\n}"]},"metadata":{},"sourceType":"module"}